test/integration/driver_install_or_update_test.go

```diff
@@ -173,3 +174,85 @@ func TestHyperKitDriverInstallOrUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestHyperKitDriverSkipUpgrade(t *testing.T) {
+	if runtime.GOOS != "darwin" {
+		t.Skip("Skip if not darwin.")
+	}
+	MaybeParallel(t)
+	tests := []struct {
+		name             string
+		path             string
+		downloadExpected bool
+	}{
+		{
+			name:             "upgrade-v1.11.0-to-current",
+			path:             filepath.Join(*testdataDir, "hyperkit-driver-version-1.11.0"),
+			downloadExpected: false,
+		},
+		{
+			name:             "upgrade-v1.2.0-to-current",
+			path:             filepath.Join(*testdataDir, "hyperkit-driver-older-version"),
+			downloadExpected: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			originalPath := os.Getenv("PATH")
+			defer func() {
+				if err := os.Setenv("PATH", originalPath); err != nil {
+					t.Errorf("failed to restore PATH: %v", err)
+				}
+			}()
+
+			dir, err := ioutil.TempDir("", tc.name)
+			if err != nil {
+				t.Fatalf("Expected to create tempdir. test: %s, got: %v", tc.name, err)
+			}
+			defer func() {
+				if err := os.RemoveAll(dir); err != nil {
+					t.Errorf("Failed to remove dir %q: %v", dir, err)
+				}
+			}()
+
+			pwd, err := os.Getwd()
+			if err != nil {
+				t.Fatalf("Error not expected when getting working directory. test: %s, got: %v", tc.name, err)
+			}
+
+			driverPath := filepath.Join(pwd, tc.path, "docker-machine-driver-hyperkit")
+			if _, err = os.Stat(driverPath); err != nil {
+				t.Fatalf("Expected driver to exist. test: %s, got: %v", tc.name, err)
+			}
+
+			// change permission to allow driver to be executable
+			if err = os.Chmod(driverPath, 0700); err != nil {
+				t.Fatalf("Expected not expected when changing driver permission. test: %s, got: %v", tc.name, err)
+			}
+
+			if err = os.Setenv("PATH", filepath.Dir(driverPath)); err != nil {
+				t.Fatalf("Unexpected error on SetEn, got: %v", err)
+			}
+			curVersion, err := version.GetSemverVersion()
+			if err != nil {
+				t.Fatalf("Expected new semver. test: %v, got: %v", tc.name, err)
+			}
+
+			if _, err = driver.AssureDriverBinary("hyperkit", dir, curVersion, true); err != nil {
```

Comment:
how about instead of using "driver.AssureDriverBinary" use
```
minikube start --download-only --driver=hyperkit
```
similar to 
https://github.com/medyagh/minikube/blob/32922f4184c31fb405fb8dab7a1d7a8ef120b47a/test/integration/aaa_download_only_test.go#L61

Reply:
When `minikube start --download-only --driver=hyperkit` actually downloads the driver it tries to set driver permission as well, asking for sudo.  `make integration` is failing in this case. 
Cannot run as `sudo make integration` too, because of 
`X Exiting due to DRV_AS_ROOT: The "hyperkit" driver should not be used with root privileges.`

Another issue is that `minikube start --download-only --driver=hyperkit` downloads not only the driver, but also preloaded-images archive and minikube.iso. It's ~700Mb per test launch.

Reply:
We have at least one integration test that runs `--download-only` (`TestDownloadOnly`) to make sure it's working properly or uses it as part of a larger test. I'm guessing all of these tests would fail on hyperkit if this doesn't work?

Reply:
Looks like my issue is specific to the external drivers only. 
`TestDownloadOnly` uses the default driver, which is usually docker. In this case minikube does not try to install an external driver binary and skips the sudo step. 

I tried to forcibly use hyperkit in `TestDownloadOnly`  passing `TEST_ARGS="-minikube-start-args --driver=hyperkit" `  and got the same error:
```
izuyev@izuyev-macbookpro --- z/minikube ‹gh_7422-reuse_hyperkit_driver* M› » make integration TEST_ARGS="-run TestDownloadOnly -minikube-start-args --driver=hyperkit"    
go test -ldflags="-X k8s.io/minikube/pkg/version.version=v1.13.1 -X k8s.io/minikube/pkg/version.isoVersion=v1.13.1 -X k8s.io/minikube/pkg/version.isoPath=minikube/iso -X k8s.io/minikube/pkg/version.gitCommitID="08da82702d38d646df313d793ab9ea0f1a4fa46d-dirty" -X k8s.io/minikube/pkg/version.storageProvisionerVersion=v3" -v -test.timeout=90m ./test/integration --tags="integration container_image_ostree_stub containers_image_openpgp go_getter_nos3 go_getter_nogcs" -run TestDownloadOnly -minikube-start-args --driver=hyperkit 2>&1 | tee "./out/testout_08da82702.txt"
ERROR: logging before flag.Parse: I1012 13:09:00.486559   47903 translate.go:64] Getting system locale failed: Could not detect Language
ERROR: logging before flag.Parse: I1012 13:09:00.486698   47903 translate.go:106] Setting Language to en-US ...
Found 12 cores, limiting parallelism with --test.parallel=6
=== RUN   TestDownloadOnly
=== RUN   TestDownloadOnly/crio
=== RUN   TestDownloadOnly/crio/v1.13.0
    aaa_download_only_test.go:65: (dbg) Run:  out/minikube start --download-only -p crio-20201012130900-47903 --force --alsologtostderr --kubernetes-version=v1.13.0 --container-runtime=crio --driver=hyperkit
Password: 

[test infinitely waits for password]
```

Reply:
I'm not super familiar with this part of the code, but do you know if we try to set the permissions before asking for root access?

My guess is that we don't need a password to set the permissions in our testing environment. If it doesn't already work like this, we could try to set the driver permissions first and then ask for root access if that fails.

Reply:
Yep, I checked CI scripts, looks like out test environment just allows sudo without a password.

Updated the test to skip if sudo needs a password.